Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,64 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func, sleeping for the corresponding number of milliseconds after failed
+// attempts, until either the function returns true or no more delays remain
+// Attempts function up to n+1 times given n delays
+template <typename FuncType, typename DelayContainerType>
+bool AttemptWithRetryDelays(const FuncType& func, const DelayContainerType& delays)
+{
+  for (const auto& delay : delays)
+  {
+    if (func())
+    {
+      return true;
+    }
+    std::this_thread::sleep_for(std::chrono::duration(delay));
+  }
+  return func();
+}
+
 // renames file srcFilename to destFilename, returns true on success
 bool Rename(const std::string& srcFilename, const std::string& destFilename)
 {
   INFO_LOG_FMT(COMMON, "Rename: {} --> {}", srcFilename, destFilename);
 #ifdef _WIN32
-  auto sf = UTF8ToTStr(srcFilename);
-  auto df = UTF8ToTStr(destFilename);
-  // The Internet seems torn about whether ReplaceFile is atomic or not.
-  // Hopefully it's atomic enough...
-  if (ReplaceFile(df.c_str(), sf.c_str(), nullptr, REPLACEFILE_IGNORE_MERGE_ERRORS, nullptr,
-                  nullptr))
-    return true;
-  // Might have failed because the destination doesn't exist.
-  if (GetLastError() == ERROR_FILE_NOT_FOUND)
-  {
-    if (MoveFile(sf.c_str(), df.c_str()))
+  // Store converted strings in local variables so cstrings don't point to temporary objects
+  const std::wstring source_wstring = UTF8ToTStr(srcFilename);
+  const std::wstring destination_wstring = UTF8ToTStr(destFilename);
+  const wchar_t* const source_ptr = source_wstring.c_str();
+  const wchar_t* const destination_ptr = destination_wstring.c_str();
+
+  // On Windows ReplaceFile can fail spuriously due to antivirus checking or other noise.
+  // Retry the operation with increasing delays, and if none of them work there's probably a
+  // persistent problem.
+  const auto attempt_func = [&, source_ptr, destination_ptr]() {
+    if (ReplaceFile(destination_ptr, source_ptr, nullptr, REPLACEFILE_IGNORE_MERGE_ERRORS, nullptr,
+                    nullptr))
+    {
       return true;
-  }
-  ERROR_LOG_FMT(COMMON, "Rename: MoveFile failed on {} --> {}: {}", srcFilename, destFilename,
-                GetLastErrorString());
+    }
+    // Might have failed because the destination doesn't exist.
+    if (GetLastError() == ERROR_FILE_NOT_FOUND)
+    {
+      return MoveFile(source_ptr, destination_ptr) != 0;
+    }
+    return false;
+  };
+  using namespace std::chrono_literals;
+  constexpr std::array delays_after_failures = {5ms, 10ms, 25ms, 50ms, 100ms, 250ms, 500ms};
```

Comment:
This seems a bit weird... I think some kind of exponential backoff would make more sense than having a manual list of delays?